### PR TITLE
Add StateTreeTasks : ShowActor, EnableActorCollision ,EnableComponentCollision and SendStateTreeEvent

### PR DESCRIPTION
**Pull request by ThibNach**
Add StateTreeTasks : ShowActor, EnableActorCollision and EnableComponentCollision

### CODE REVIEW

**Review by Frickerson — DISMISSED** [review]

**Review by Frickerson — DISMISSED** [review]

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskSendStateTreeEvent.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include <Components/StateTreeComponent.h>
+#include <StateTreeExecutionContext.h>
+
+EStateTreeRunStatus FAIExtStateTreeTaskSendStateTreeEvent::EnterState( FStateTreeExecutionContext & context, const FStateTreeTransitionResult & transition ) const
+{
+    const auto & instance_data = context.GetInstanceData< FInstanceDataType >( *this );
+
+    if ( instance_data.StateTreeComponent != nullptr )
+    {
+        instance_data.StateTreeComponent->SendStateTreeEvent( instance_data.EventTag );
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskSendStateTreeEvent.cpp]: you should expose also the payload and the origin

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskEnableActorCollision.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include "StateTree/Tasks/AIExtStateTreeTaskEnableActorCollision.h"
+
+#include <StateTreeExecutionContext.h>
+
+EStateTreeRunStatus FAIExtStateTreeTaskEnableActorCollision::EnterState( FStateTreeExecutionContext & context, const FStateTreeTransitionResult & transition ) const
+{
+    const auto & instance_data = context.GetInstanceData< FInstanceDataType >( *this );
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskEnableActorCollision.cpp:9]: use `TRACE_CPUPROFILER_EVENT_SCOPE_STR( __FUNCTION__ );` in each function of each task

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskSendStateTreeEvent.cpp (hunk 1 above)
Comment (ThibNach) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskSendStateTreeEvent.cpp]: DOne

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskEnableActorCollision.cpp (hunk 2 above)
Comment (ThibNach) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskEnableActorCollision.cpp:9]: Done

**Review by TheEmidee — APPROVED** [review]

**Pull request merged**